src/System.Memory/src/System/Buffers/ReadOnlySequence_helpers.cs

```diff
@@ -11,207 +12,171 @@ namespace System.Buffers
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         internal bool TryGetBuffer(SequencePosition start, SequencePosition end, out ReadOnlyMemory<T> data, out SequencePosition next)
         {
-            if (start.Segment == null)
+            SequenceType type;
+            next = default;
+            int startIndex = 0, endIndex = 0;
+            object startSegment = start.Segment;
+
+            if (startSegment == null)
             {
-                data = default;
-                next = default;
-                return false;
+                type = SequenceType.Empty;
```

Comment:
Does it make sense to merge this if into GetTypeAndIndices?

Reply:
Done